### DISCUSSION TITLE
Minimal overhead mechanism for feeding body data if buffers already available

### DISCUSSION THREAD

**Discussion by joshi-mohit** [corazawaf/coraza#832] [Q&A] [answered]
I already have the request / response body in local buffers and want to feed Coraza for analysis. Is there a way to feed this to ProcessRequestBody/ProcessResponseBody with minimal overhead of copying in separate buffers. Does the coraza internally changes these buffer so a copy is must

**Comment by jcchavezs** [comment]
Coraza does not mutate the body, we have this method to be able to reuse a
slice of bytes
https://github.com/corazawaf/coraza/blob/main/internal/corazawaf/transaction.go#L787.
what connector are you using?

**Reply by joshi-mohit** [reply]
Thanks for this... I did see this but could it is making a copy. I have started experimenting with Coraza so do not have a good understanding here.
  ... it is a proprietary connectors sending data over REST interface . But in the processing pipeline I am deep to use the buffer to reuse the buffer coming back from coraza, so was just looking for some interface in which a existing data-buffers could be reused by coraza  ... I am just experimenting a few things here 
Thanks a lot for quick response here

**Comment by jptosso** [comment]
You can flush your body while feeding coraza then replace your reader with coraza buffer reader. That way you avoid maintaining the body in memory twice.
Coraza reader behaves different depending on the configuration, you might get a pointer to the memory buffer or a pointer to the file reader.
Coraza doesn't change the request or response bodies. It is safe to extract it from coraza.

**Comment by jcchavezs** [comment]
Only caveat on using coraza buffets is that coraza buffers the body up to a
limit and not entire body.

**Comment by jcchavezs** [comment]
Cc @anuraaga

**Comment by jcchavezs** [comment, marked as answer]
Maybe be could expose a `unsafeUseRequestBody` and accept a body buffer
because we also reject based on length.

**Reply by joshi-mohit** [reply]
Once I get a better can pick it up

**Comment by jcchavezs** [comment]
I think some example code could work, what would you expect from such a
method to accept as parameter?
